Remove old Type based backend extensions

li-roy opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22009 Remove old Type based backend extensions**

Extension backends should use ATenDispatch.h now.

Removed some tests because we shouldn't be overriding functions that don't have dispatch for backends.

Differential Revision: [D15914755](https://our.internmc.facebook.com/intern/diff/D15914755)

Commit:
Remove old Type based backend extensions

Commit:
Update on "Remove old Type based backend extensions"

Remove old Type based backend extensions

gh-metadata: pytorch pytorch 22009 gh/li-roy/34/head

gchanan commented:
looks like some tests are failing.

gchanan reviewed (approved):
this looks fine, but I don't know what "Removed some tests because we shouldn't be overriding functions that don't have dispatch for backends" is supposed to mean.

Commit:
Update on "Remove old Type based backend extensions"

Remove old Type based backend extensions

gh-metadata: pytorch pytorch 22009 gh/li-roy/34/head

Commit:
Update on "Remove old Type based backend extensions"

Remove old Type based backend extensions

gh-metadata: pytorch pytorch 22009 gh/li-roy/34/head

gchanan commented:
is this going to be merged?

Commit:
Update on "Remove old Type based backend extensions"

Remove old Type based backend extensions

gh-metadata: pytorch pytorch 22009 gh/li-roy/34/head
